adapters/repos/db/vector/hnsw/maintenance_cycles.go

```diff
@@ -0,0 +1,146 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2023 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package hnsw
+
+import (
+	"context"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/weaviate/weaviate/entities/cyclemanager"
+	"golang.org/x/sync/errgroup"
+)
+
+type MaintenanceCycles struct {
+	commitLogMaintenance cyclemanager.CycleManager
+	tombstoneCleanup     cyclemanager.CycleManager
+
+	lock sync.Mutex
+}
+
+func (c *MaintenanceCycles) Init(
+	commitlogMaintenanceTicker cyclemanager.CycleTicker,
+	tombstoneCleanupTicker cyclemanager.CycleTicker,
+) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	if c.commitLogMaintenance == nil &&
+		c.tombstoneCleanup == nil {
+		c.commitLogMaintenance = cyclemanager.NewMulti(commitlogMaintenanceTicker)
+		c.tombstoneCleanup = cyclemanager.NewMulti(tombstoneCleanupTicker)
+
+		c.commitLogMaintenance.Start()
```

Comment:
c.commitLogMaintenance or c.tombstoneCleanup one of them could be nil
I think we need to check for nil pointer before calling start